editor/plugins/mesh_editor_uv_tools.cpp

```diff
@@ -0,0 +1,293 @@
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "mesh_editor_uv_tools.h"
+
+#include "editor/editor_node.h"
+#include "editor/editor_string_names.h"
+#include "editor/editor_undo_redo_manager.h"
+// #include "editor/multi_node_edit.h"
+// #include "editor/plugins/node_3d_editor_plugin.h"
+#include "editor/themes/editor_scale.h"
+// #include "scene/3d/navigation_region_3d.h"
+// #include "scene/3d/physics/collision_shape_3d.h"
+// #include "scene/3d/physics/physics_body_3d.h"
+// #include "scene/3d/physics/static_body_3d.h"
+#include "scene/gui/aspect_ratio_container.h"
+// #include "scene/gui/box_container.h"
+#include "scene/gui/dialogs.h"
+// #include "scene/gui/menu_button.h"
+// #include "scene/gui/spin_box.h"
+// #include "scene/resources/3d/concave_polygon_shape_3d.h"
+// #include "scene/resources/3d/convex_polygon_shape_3d.h"
+#include "scene/resources/3d/primitive_meshes.h"
+
+struct MeshInstance3DEditorEdgeSort {
+	Vector2 a;
+	Vector2 b;
+
+	static uint32_t hash(const MeshInstance3DEditorEdgeSort &p_edge) {
+		uint32_t h = hash_murmur3_one_32(HashMapHasherDefault::hash(p_edge.a));
+		return hash_fmix32(hash_murmur3_one_32(HashMapHasherDefault::hash(p_edge.b), h));
+	}
+
+	bool operator==(const MeshInstance3DEditorEdgeSort &p_b) const {
+		return a == p_b.a && b == p_b.b;
+	}
+
+	MeshInstance3DEditorEdgeSort() {}
+	MeshInstance3DEditorEdgeSort(const Vector2 &p_a, const Vector2 &p_b) {
+		if (p_a < p_b) {
+			a = p_a;
+			b = p_b;
+		} else {
+			b = p_a;
+			a = p_b;
+		}
+	}
+};
+
+void MeshEditorUvTools::create_uv_lines(Ref<Mesh> mesh, int p_layer) {
```

Comment:
The class should be named MeshEditorUVTools (UV instead of Uv), as per Godot's naming conventions.